Cpp_test/lib/ipc-common.h

```diff
@@ -0,0 +1,48 @@
+#ifndef IPC_COMMON_H_
+#define IPC_COMMON_H_
+
+#include <iostream>
+#include <utility>
+
+enum class IPCProtocol
+{
+	NONE, QUEUE, HELP, PIPE, SHARE, 
+};
+
+
+enum class IPCMode
+{
+	SEND_MODE, RECEIVE_MODE
+};
+
+
+struct ipc_info{
+    IPCMode mode{};
+    IPCProtocol protocol{};
+	char *file_name;
+    char *method_name;
+	int fflag;
+	int qflag;
+	int pflag;
+	int sflag;
+    int hflag;
+	
+};
+
+
+class IPC
```

Comment:
Generally it's better to name the file the same as the class and not group many common items together in one file. If this was a C# project, all enums, structs and classes would have their own separate file, but maybe it's different in C++. Someone more experienced in C++ projects can clarify. In any case, I would avoid putting main classes in a common file anyway. Maybe could rename this as "ipc.h" or create a new file for the ipc class.

Reply:
yeah, I rename the file to "ipc.h"